[Android] App is getting crashed. allowInsecureConnections is null

## Issue

In createConnectionBuilder function allowInsecureConnections coming as null. Android app is getting crashed because of this. But if I try 2-3 times it works. Please help my app is getting crashed in production.
Stack trace:
```
 java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=0, result=-1, data=Intent { dat=com.livspace.canvas.android:///login?code=pZJYUn2Q4zJ3xQIngXEexFP5YHJ9NL87zFqih63csmk.wtJ4Ql6-2BTHJfZbWKs12YlF27Nqoq1uT2ViNFmQgjk&scope=openid offline offline_access&state=2JjhOtCB-80s4UfomoVIkQ (has extras) }} to activity {com.livspace.canvas.android/com.livspace.canvas.android.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Boolean.equals(java.lang.Object)' on a null object reference
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4415)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4459)
        at android.app.ActivityThread.-wrap19(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1738)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:192)
        at android.app.ActivityThread.main(ActivityThread.java:6754)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Boolean.equals(java.lang.Object)' on a null object reference
        at com.rnappauth.RNAppAuthModule.createConnectionBuilder(RNAppAuthModule.java:538)
        at com.rnappauth.RNAppAuthModule.onActivityResult(RNAppAuthModule.java:317)
        at com.facebook.react.bridge.ReactContext.onActivityResult(ReactContext.java:277)
        at com.facebook.react.ReactInstanceManager.onActivityResult(ReactInstanceManager.java:704)
        at com.facebook.react.ReactActivityDelegate.onActivityResult(ReactActivityDelegate.java:129)
        at com.facebook.react.ReactActivity.onActivityResult(ReactActivity.java:75)
        at android.app.Activity.dispatchActivityResult(Activity.java:7433)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4411)


```
---

## Environment

* **Your Identity Provider**: Hydra
* **Platform that you're experiencing the issue on**: Android
* **Are you using Expo?** No

--- a/android/src/main/java/com/rnappauth/RNAppAuthModule.java
+++ b/android/src/main/java/com/rnappauth/RNAppAuthModule.java
@@ -60,7 +60,7 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
 
     private final ReactApplicationContext reactContext;
     private Promise promise;
-    private Boolean dangerouslyAllowInsecureHttpRequests;
+    private boolean dangerouslyAllowInsecureHttpRequests;
     private String clientAuthMethod = "basic";
     private Map<String, String> registrationRequestHeaders = null;
     private Map<String, String> authorizationRequestHeaders = null;
@@ -84,7 +84,7 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
         final String clientId,
         final ReadableArray scopes,
         final ReadableMap serviceConfiguration,
-        final Boolean dangerouslyAllowInsecureHttpRequests,
+        final boolean dangerouslyAllowInsecureHttpRequests,
         final ReadableMap headers,
         final Promise promise
     ) {
@@ -147,7 +147,7 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
         final String tokenEndpointAuthMethod,
         final ReadableMap additionalParameters,
         final ReadableMap serviceConfiguration,
-        final Boolean dangerouslyAllowInsecureHttpRequests,
+        final boolean dangerouslyAllowInsecureHttpRequests,
         final ReadableMap headers,
         final Promise promise
     ) {
@@ -217,7 +217,7 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
             final ReadableMap serviceConfiguration,
             final Boolean usePKCE,
             final String clientAuthMethod,
-            final Boolean dangerouslyAllowInsecureHttpRequests,
+            final boolean dangerouslyAllowInsecureHttpRequests,
             final ReadableMap headers,
             final Promise promise
     ) {
@@ -297,7 +297,7 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
             final ReadableMap additionalParameters,
             final ReadableMap serviceConfiguration,
             final String clientAuthMethod,
-            final Boolean dangerouslyAllowInsecureHttpRequests,
+            final boolean dangerouslyAllowInsecureHttpRequests,
             final ReadableMap headers,
             final Promise promise
     ) {
@@ -707,10 +707,10 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
     /*
      *  Create appropriate connection builder based on provided settings
      */
-    private ConnectionBuilder createConnectionBuilder(Boolean allowInsecureConnections, Map<String, String> headers) {
+    private ConnectionBuilder createConnectionBuilder(boolean allowInsecureConnections, Map<String, String> headers) {
         ConnectionBuilder proxiedBuilder;
 
-        if (allowInsecureConnections.equals(true)) {
+        if (allowInsecureConnections) {
             proxiedBuilder =UnsafeConnectionBuilder.INSTANCE;
         } else {
             proxiedBuilder = DefaultConnectionBuilder.INSTANCE;

